alacritty_terminal/src/term/mod.rs

```diff
@@ -1875,8 +1894,20 @@ impl<T: EventListener> Handler for Term<T> {
                     let template = Cell { bg: template.bg, ..Cell::default() };
                     self.grid.clear_viewport(template);
                 }
+
+                self.selection =
+                    self.selection.take().filter(|s| !s.intersects_range(0..=last_line));
+            },
+            ansi::ClearMode::Saved => {
+                let last_scrollback_line = max(self.grid.len() - 1, last_line + 1);
```

Comment:
Why are you `max`ing things here? Just in case there is no history?

Can we maybe just add a conditional to check that and skip this branch completely in that case? That way we don't have to do any expensive clearing operations at all.